app/services/IdentityMatchService.scala

```diff
@@ -20,22 +20,24 @@ import config.AppConfig
 import connectors.IdentityMatchConnector
 import exceptions.{InvalidIdMatchRequest, LimitException}
 import javax.inject.Inject
-import models.api1585.{IdMatchApiError, IdMatchApiResponseSuccess, DownstreamServerError}
+import models.api1585.{DownstreamServerError, IdMatchApiError, IdMatchApiResponseSuccess}
 import models.{BinaryResult, IdMatchRequest, IdMatchResponse}
 import play.api.Logging
 import repositories.IndividualCheckRepository
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Session
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class IdentityMatchService @Inject()(val connector: IdentityMatchConnector,
                                      val repository: IndividualCheckRepository,
                                      val appConfig: AppConfig) extends Logging {
 
-  def matchId(request: IdMatchRequest)(implicit ec: ExecutionContext): Future[Either[IdMatchApiError, IdMatchResponse]] = {
+  def matchId(request: IdMatchRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[IdMatchApiError, IdMatchResponse]] = {
 
     limitedMatch(request).recoverWith {
       case _: InvalidIdMatchRequest =>
-        logger.warn(s"[IdMatchService] unable to send request, failed validation")
+        logger.warn(s"[Session ID: ${Session.id(hc)}] unable to send request, failed validation")
```

Comment:
@BriWak Could we add some further logs into the methods
`clearCounter`
`limitedMatch`

In particular:
- Lock cleared
- max attempts exceeded with the current count
- matched
- not matched

all with the session id